### PR TITLE
fix: fix error when msg.space is empty

### LINKED ISSUE #139: Error: Errors found in subgraphRequest: URL: https://hub.snapshot.org/graphql, Status: 400,  Response: [...
Sentry Issue: [SNAPSHOT-RELAYER-1](https://snapshot-labs.sentry.io/issues/4401679917/?referrer=github_integration)

```
Error: Errors found in subgraphRequest: URL: https://hub.snapshot.org/graphql, Status: 400,  Response: [{"message":"String cannot represent a non string value: undefined","locations":[{"line":1,"column":20}]}]
  File "/workspace/node_modules/@snapshot-labs/snapshot.js/dist/snapshot.cjs.js", line 4414, in Object.<anonymous>
    throw new Error("Errors found in subgraphRequest: URL: " + url + ", Status: " + res.status + ",  Response: " + JSON. {snip}
  File "/workspace/node_modules/@snapshot-labs/snapshot.js/dist/snapshot.cjs.js", line 82, in step
    op = body.call(thisArg, _);
  File "/workspace/node_modules/@snapshot-labs/snapshot.js/dist/snapshot.cjs.js", line 63, in Object.next
    function verb(n) { return function (v) { return step([n, v]); }; }
  File "/workspace/node_modules/@snapshot-labs/snapshot.js/dist/snapshot.cjs.js", line 53, in fulfilled
    function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
  File "<anonymous>", in runMicrotasks
...
(1 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by wa0x6e**
## 🧿 Current issues / What's wrong ?

When an user is submitting a payload with an empty `msg.space`, the subgraph query using that space as argument is failing with the error https://snapshot-labs.sentry.io/issues/4401679917/?referrer=github_integration

## 💊 Fixes / Solution

Fix #139 
Fix [SNAPSHOT-RELAYER-1](https://snapshot-labs.sentry.io/issues/4401679917/?referrer=github_integration)

## 🚧 Changes

- Check and return a 400 error when user is submitting an invalid payload (missing msg.space)

## 🛠️ Tests

- N/A

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -66,14 +66,21 @@ router.get('/api/messages/:hash', async (req, res) => {
 });
 
 router.post('/', async (req, res) => {
+  const msg = req.body.data.message;
+
   if (!req.body.data || !req.body.data.message) {
```
Comment (ChaituVR) [inline src/api.ts:71]:
```suggestion
  const msg = req.body.data?.message;

  if (!msg) {
```

**Review by ChaituVR — APPROVED** [review]
utAck

File: src/api.ts [hunk 2]
```diff
@@ -72,8 +72,14 @@ router.post('/', async (req, res) => {
     });
   }
 
+  const msg = req.body.data.message;
+  if (!req.body.data.types.Space && !msg.settings && !msg.space) {
```
Comment (ChaituVR) [inline src/api.ts:76]: maybe better to check schema like on sequencer? 🤔

Comment (wa0x6e) [reply src/api.ts:77]: We could certainly do that, it's outside this PR scope. I'll open another issue for that

**Pull request merged**